pyeapi/api/interfaces.py

```diff
@@ -407,8 +411,55 @@ def delete(self, name):
             NotImplementedError: Deleting  Ethernet interfaces is not supported
 
         """
+        ver = self.node.version_number.split('.')
+        if int(ver[0]) >= 4 and int(ver[1]) >= 15:
+            return self.configure('no interface %s' % name)
```

Comment:
Would it make sense to see if the version isn't 4.15 and raise an exception? Then put the return self.configure at the bottom?  The way it's written now doesn't look pythonic.